service/resource/cloudformationv2/main_stack.go

```diff
@@ -18,31 +18,36 @@ import (
 
 func newMainStack(customObject v1alpha1.AWSConfig) (StackState, error) {
 	stackName := keyv2.MainGuestStackName(customObject)
-	workers := len(customObject.Spec.AWS.Workers)
-	var workerImageID, workerInstanceType string
-	// FIXME: the imageID should not depend on the number of workers.
+	workerCount := keyv2.WorkerCount(customObject)
```

Comment:
As well as using the ImageID I also did some boyscouting to use keyv2 helpers in more places.